### PR TITLE
Code for Post subscription Usage

### PR DESCRIPTION
**Pull request by vivyzer**

### CODE REVIEW

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 1]
```diff
@@ -17,6 +17,24 @@
...
 package com.ning.billing.recurly;
 
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Objects;
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.io.CharSource;
+import com.google.common.io.Resources;
+import com.google.common.net.HttpHeaders;
+import com.ning.billing.recurly.model.*;
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: We'd prefer to stay away from * imports

Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: Could you just add the import for the `Usage` class instead?

File: src/main/java/com/ning/billing/recurly/model/Usage.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        if (updatedAt != null ? !updatedAt.equals(usage.updatedAt) : usage.updatedAt != null) return false;
+        if (billedAt != null ? !billedAt.equals(usage.billedAt) : usage.billedAt != null) return false;
+        if (usageType != null ? !usageType.equals(usage.usageType) : usage.usageType != null) return false;
+        if (unitAmountInCents != null ? !unitAmountInCents.equals(usage.unitAmountInCents) : usage.unitAmountInCents != null)
+            return false;
+        return usagePercentage != null ? usagePercentage.equals(usage.usagePercentage) : usage.usagePercentage == null;
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/model/Usage.java]: Could you change this to use `Objects.hashCode` like we do in the other classes?
https://github.com/killbilling/recurly-java-library/blob/master/src/main/java/com/ning/billing/recurly/model/Account.java#L367

File: src/main/java/com/ning/billing/recurly/model/Usage.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        result = 31 * result + (unitAmountInCents != null ? unitAmountInCents.hashCode() : 0);
+        result = 31 * result + (usagePercentage != null ? usagePercentage.hashCode() : 0);
+        return result;
+    }
+
+    public Integer getAmount() {
+        return amount;
+    }
+
+    public void setAmount(Object amount) {
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/model/Usage.java]: Could you mark these setter params as `final`?

**Comment by bhelx** [comment]
@vivyzer thank you so much for contributing this. It's great work! I left a few small comments. Let me know if you have any questions.

**Comment by vivyzer** [comment]
@bhelx Thank you for your comments. Updated the code. Let me know if this looks fine.

File: src/main/java/com/ning/billing/recurly/model/Usage.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Usage usage = (Usage) o;
+
+        if (!amount.equals(usage.amount)) return false;
+        if (merchantTag != null ? !merchantTag.equals(usage.merchantTag) : usage.merchantTag != null) return false;
+        if (recordingAt != null ? !recordingAt.equals(usage.recordingAt) : usage.recordingAt != null) return false;
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/model/Usage.java]: Could you change this to use compareTo for the datetimes like we do in the other models?
https://github.com/killbilling/recurly-java-library/blob/master/src/main/java/com/ning/billing/recurly/model/Account.java#L320

**Comment by bhelx** [comment]
@vivyzer Just have one other comment then it's good to go. Thanks for being so attentive.

**Comment by vivyzer** [comment]
@bhelx done. Thank you !!

File: src/main/java/com/ning/billing/recurly/model/Usage.java (hunk 4 above)
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/model/Usage.java]: @vivyzer you missed `recordingAt` I think

**Comment by vivyzer** [comment]
@bhelx Sorry for missing that field. Checked in again. Thanks.

**Comment by bhelx** [comment]
:+1: Thanks a lot. If you find anything else missing feel free to file an issue and I can add it. Also happy to review any other PRs you have.

**Pull request merged**